samples/bluetooth/direct_test_mode/src/dtm.c

```diff
@@ -446,18 +444,26 @@ static void radio_gpio_pattern_clear(void)
 
 static void antenna_radio_pin_config(void)
 {
-	const uint32_t *pin = dtm_hw_radion_antenna_pin_array_get();
+	const uint8_t *pin = dtm_hw_radio_antenna_pin_array_get();
+
+	for (size_t i = 0; i < DTM_HW_MAX_DFE_GPIO; i++) {
+		uint32_t pin_value = (pin[i] == DTM_HW_DFE_PSEL_NOT_SET) ?
+				     DTM_HW_DFE_GPIO_PIN_DISCONNECT : pin[i];
 
-	for (uint8_t i = 0; i < dtm_radio_antenna_pin_array_size_get(); i++) {
-		NRF_RADIO->PSEL.DFEGPIO[i] = pin[i];
+		nrf_radio_dfe_pattern_pin_set(NRF_RADIO,
+					      pin_value,
+					      i);
 	}
 }
 
 static void switch_pattern_set(void)
 {
-	/* Set antenna for the guard period and for the reference period. */
-	NRF_RADIO->SWITCHPATTERN = 1;
-	NRF_RADIO->SWITCHPATTERN = 1;
+	uint8_t pdu_antenna = dtm_hw_radio_pdu_antenna_get();
+	/* Set antenna for the PDU, guard period and for the reference period.
+	 * The same antenna is used for guard and reference period like for the PDU.
+	 */
+	NRF_RADIO->SWITCHPATTERN = pdu_antenna;
+	NRF_RADIO->SWITCHPATTERN = pdu_antenna;
```

Comment:
The BLE Spec is unclear here. I took an assumption that we should use the PDU antenna also for the Reference period transmission and reception. I added information about this in the sample documentation. We should clarify this in the BLE Spec because the DTM command can set antenna number between 0x01-0x4b but the minimal antenna count for CTE is 2. The DTM HCI command requires at least 2 antennas in a switch pattern